src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -239,7 +245,7 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     new JesBackendCall(this, workflowDescriptor, key, locallyQualifiedInputs, abortRegistrationFunction)
   }
 
-  override def engineFunctions: WdlStandardLibraryFunctions = new JesEngineFunctionsWithoutCallContext()
+  override def engineFunctions(interface: IOInterface): WdlStandardLibraryFunctions = new JesEngineFunctionsWithoutCallContext(interface)
```

Comment:
I think yes :s, the whole problem here is that we need to be able to read files when we evaluate expressions in the workflow declaration, which happens when the workflow descriptor is created.
This means that the workflow descriptor needs something to pass to the engineFunctions so they can read files (for `read_lines` etc...). However we don't want to introduce backend specific terminology/logic at the workflow descriptor level, which is why IOInterface is completely generic.
Maybe with some kind of [T] parameter it would be possible to make sure that we pass the right interface.